### DISCUSSION TITLE
GPU not really get used just RAM and CPU

### DISCUSSION THREAD

**Discussion by Schakuun** [voicepaw/so-vits-svc-fork#372] [General]
I have started several training sessions and noticed that the system primarily uses RAM and CPU resources while the GPU is hardly utilized. Even when I adjusted the batch size to 64, which resulted in 90-95% RAM and CPU utilization, the GPU only runs at 15-30% usage. Is this intentional, or can I utilize the GPU more effectively?

Sorry if this question seems too basic.

For my setup, I am using:

    OS: Windows 11
    Env: Conda
    GPU: RTX 3090 Trinity OC 24GB
    CPU: i7-10700K
    RAM: 32GB 2400MHz

**Comment by hypersniper05** [comment]
same here

*cuda 11.8

**Comment by Lordmau5** [comment]
In Task Manager in the GPU tab, make sure you select Cuda for one of the views
![image](https://user-images.githubusercontent.com/1345036/232881021-852342a4-e3c8-4db2-9855-312884069939.png)

The 3D, Copy and Video Encode / Decode aren't being used for CUDA related things

Does that graph show usage when you're training?

(If you can't select Cuda in that or the other dropdowns make sure you have the CUDA toolkit installed and restart your PC afterwards)

**Reply by Schakuun** [reply]
Thanks for the information, unfortunaly it uses 0% of Cuda.
BatchSize is 32 as seen in config file. 

Dataset:
- 770 AudioFiles

![So-Vits-Bug-Cuda](https://user-images.githubusercontent.com/118875624/232980977-a6d3bad1-6430-4baa-96c3-a864bf25dc34.jpg)

**Reply by Schakuun** [reply]
Found this after restarting the Process. It seems Conda Env cant use the GPU.
![grafik](https://user-images.githubusercontent.com/118875624/232988597-309b9e67-a98a-4199-8fc8-30dd427bea62.png)

**Reply by Schakuun** [reply]
I have resolved the issue by reinstalling PyTorch-Cuda 11.8 in the conda environment "so-vits-svc-fork". The command I used was: "conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia".

After completing the installation, I was able to run "svc pre-hubert" again. The only missing dependencies were "chardet" and "cchardet", which I installed using the following commands: "pip install chardet" and "pip install cchardet".

After installing these dependencies, everything worked as expected, and the GPU was fully utilized.

**Comment by pxtchxs** [comment]
yo i am having this same issue:

INFO: GPU available: False, used: False
           INFO     [17:43:27] GPU available: False, used: False                                                                                               rank_zero.py:48
INFO: TPU available: False, using: 0 TPU cores
           INFO     [17:43:27] TPU available: False, using: 0 TPU cores                                                                                        rank_zero.py:48
INFO: IPU available: False, using: 0 IPUs
           INFO     [17:43:27] IPU available: False, using: 0 IPUs                                                                                             rank_zero.py:48
INFO: HPU available: False, using: 0 HPUs
           INFO     [17:43:27] HPU available: False, using: 0 HPUs

when i try to train. i have evga 3090ti and i9-13900k. miniconda. cuda 11.8. i tried the "conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia". "pip install chardet" and "pip install cchardet". commands and they installed, but still didnt fix the issue. please pleas please how do i fix this??

**Reply by Schakuun** [reply]
Before i did reintall the pytorch in the Conda ENV i installed "Cuda-toolkit", 
Heres the link:
https://developer.nvidia.com/cuda-downloads?target_os=Windows&target_arch=x86_64&target_version=11&target_type=exe_local

Additionally i changed a Setting in the Windows System Setting go to: Settings -> System -> Display -> Graphics settings (down the bottom of page) -> Hardware-accelerated GPU scheduling -> Switch to "Off. 

Hopefully that helps u.

**Reply by hypersniper05** [reply]
I reinstalled everything in the anaconda env and it works now :) thanks

**Reply by hypersniper05** [reply]
I reinstalled everything in the anaconda env and it works now :) thanks

**Comment by toroskuy** [comment]
![image](https://github.com/voicepaw/so-vits-svc-fork/assets/83744917/a5ac143d-a346-4779-aa79-db1f66ecdcaf)
did all of the steps you said.. still nothing

**Reply by toroskuy** [reply]
is cuda 12 supported?

**Reply by jiaxinf** [reply]
check out my solution here https://github.com/voicepaw/so-vits-svc-fork/discussions/779#discussioncomment-6224341

**Comment by hypersniper05** [comment]
Reinstall everything including the anaconda env. That's what did it for me
